File: Lect-1-C/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

void print_row(int total_bricks, int row_number);

int main(void)
{
    // Prompt the user for the pyramid's height
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1 || n > 8);

    // Print a pyramid of that height
    for (int i = 0; i < n; i++)
    {
        // Print row of bricks
        print_row(n, i + 1);
    }
}

void print_row(int total_bricks, int row_number)
{
    // Print spaces before the bricks
    for (int i = 0; i < total_bricks - row_number; i++)
    {
        printf(" ");
    }

    // Print bricks
    for (int i = 0; i < row_number; i++)
    {
        printf("#");
    }

    printf("\n");
}

File: Lect-6-Python/sentimental-mario-less/mario.py
def print_row(total_bricks, row_number):
    # Print spaces before the bricks
    for i in range(total_bricks - row_number):
        print(" ", end="")

    # Print bricks
    for i in range(row_number):
        print("#", end='')

    print()


def main():
    # Prompt the user for the pyramid's height
    while True:
        try:
            n = int(input("Height: "))
            if 1 <= n <= 8:
                break
            else:
                print('Height must be between 1 and 8')
        except ValueError:
            print("Invalid input. Please enter an integer")

    for i in range(n):
        print_row(n, i + 1)


if __name__ == "__main__":
    main()
